planner/core/indexmerge_path.go

```diff
@@ -677,16 +655,18 @@ func jsonArrayExpr2Exprs(sctx sessionctx.Context, jsonArrayExpr expression.Expre
 	return exprs, true
 }
 
-func jsonValue2Expr(v types.BinaryJSON) (expression.Expression, bool) {
-	if v.TypeCode != types.JSONTypeCodeInt64 {
-		// only support INT now
-		// TODO: support more types
+func jsonValue2Expr(sctx sessionctx.Context, v types.BinaryJSON, targetType *types.FieldType) (expression.Expression, bool) {
+	convertFunc := expression.ConvertJSON2Tp(targetType.EvalType())
+	if convertFunc == nil {
+		return nil, false
+	}
+	datum, err := convertFunc(sctx.GetSessionVars().StmtCtx, v, targetType)
```

Comment:
we can not use `sctx.GetSessionVars().StmtCtx` directly here. If the sql mode is `null` some error is ignored

Reply:
please ref the usage in expression package